features/topology-map/plugins/org.opennms.features.topology.plugins.topo.pathoutage/src/test/java/org/opennms/features/topology/plugins/topo/pathoutage/TestNodeGenerator.java

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.features.topology.plugins.topo.pathoutage;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.monitoringLocations.OnmsMonitoringLocation;
+
+import com.google.common.collect.Lists;
+
+/**
+ * This class with is responsible for generation of test {@link OnmsNode} objects
+ */
+class TestNodeGenerator {
+	/**
+	 * This method generates 9 {@link OnmsNode} objects for testing of
+	 * {@link PathOutageProvider} and {@link PathOutageStatusProvider}
+	 * @param location {@link OnmsMonitoringLocation} or null
+	 * @return Resulting map - values are nodes' levels in the path outage hierarchy
+	 */
+	public static Map<OnmsNode, Integer> generateNodes(OnmsMonitoringLocation location) {
+		Map<OnmsNode, Integer> nodesMap = new HashMap<>();
+
+		OnmsNode node0 = new OnmsNode();
+		node0.setId(1);
+		node0.setLabel("Node-0");
+		if (location != null) {
+			node0.setLocation(location);
+		}
+		nodesMap.put(node0, 0);
```

Comment:
There should be a method `createNode(int id, String label, String location)` and should be used instead.